src/NanoMsg.cpp

```diff
@@ -0,0 +1,49 @@
+#include "NanoMsg.h"
+#include <cstring>
+#include <stdexcept>
+#include <g3log/g3log.hpp>
+/**
+ * Constructor for copying a packet hash vector into a nanomsg buffer
+ */
+NanoMsg::NanoMsg(PacketHashVector& data) : sent(false) {
```

Comment:
@KjellKod @john-gress i'll keep it as is for now, and if we see the need for it I can add the user pointer api.

if anyone is curious you can look at http://nanomsg.org/development-bindings.html, Sending and Receive Messages, and also 

http://nanomsg.org/v0.8/nn_send.3.html
http://nanomsg.org/v0.8/nn_recv.3.html